### PR TITLE
Final

### PR DESCRIPTION
**Pull request by stefangusa**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.DS_Store
+
```
Comment (serban21) [inline .gitignore]: Don't push this into the master. Use git exclude file, see https://www.jerriepelser.com/blog/exlude-git-files-locally/

Comment (stefangusa) [reply .gitignore]: got it, corect

File: actions/lib/action.py [hunk 2]
```diff
@@ -56,11 +57,61 @@ def __init__(self, config):
...
         if region:
             self.credentials['region'] = region
 
+        self.cross_roles_arns = self._get_cross_account_ids()
+
         self.resultsets = ResultSets()
 
+    def _get_cross_account_ids(self):
+        cross_roles_arns = {}
+        cross_roles_arns_list = self.config.get('roles_arns') or []
```
Comment (serban21) [inline actions/lib/action.py]: Should work with `self.config.get('roles_arns', [])`

File: actions/lib/action.py [hunk 3]
```diff
@@ -56,11 +57,61 @@ def __init__(self, config):
...
+
+        for arn in cross_roles_arns_list:
+            cross_roles_arns[arn.split(':')[4]] = arn
```
Comment (serban21) [inline actions/lib/action.py]: Why not use a dict comprehension here?

File: actions/lib/action.py [hunk 4]
```diff
@@ -56,11 +57,61 @@ def __init__(self, config):
...
+
+        return cross_roles_arns
+
+    def _check_queue_if_url(self, queue_url):
+        reg = re.compile(r"https?://")
+        if reg.match(queue_url[:7]) or reg.match(queue_url[:8]):
```
Comment (serban21) [inline actions/lib/action.py]: Use `.startswith("http://") or .startswith("https://")`

Also, I don't think we really need a function just for this. See also comment on `_get_account_id_from_queue_url`

File: actions/lib/action.py [hunk 5]
```diff
@@ -56,11 +57,61 @@ def __init__(self, config):
...
+
+    def _get_account_id_from_queue_url(self, queue_url):
+        if self._check_queue_if_url(queue_url):
+            return queue_url.split('/')[3]
+        return self.account_id
+
+    def _get_region_from_queue_url(self, queue_url):
+        if self._check_queue_if_url(queue_url):
+            return queue_url.split('.')[1]
+        return self.aws_region
```
Comment (serban21) [inline actions/lib/action.py]: These 2 function should be one, that returns a tuple of `account_id, region`

File: sensors/sqs_sensor.py [hunk 6]
```diff
@@ -98,50 +123,114 @@ def _may_setup_sqs(self):
...
+            else:
+                return sqs_res.Queue(queue)
         except ClientError as e:
             if e.response['Error']['Code'] == 'AWS.SimpleQueueService.NonExistentQueue':
-                self._logger.warning("SQS Queue: %s doesn't exist, creating it.", queueName)
-                return self.sqs_res.create_queue(QueueName=queueName)
+                if not self._check_queue_if_url(queue):
+                    self._logger.warning("SQS Queue: %s doesn't exist, creating it.", queue)
+                    return sqs_res.create_queue(QueueName=queue)
+                else:
```
Comment (serban21) [inline sensors/sqs_sensor.py:212]: You should create queue on this branch, too

**Review by serban21 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**